src/any_agent/tools/mcp/frameworks/agno.py

```diff
@@ -15,44 +23,39 @@
     mcp_available = True
 
 
-class AgnoMCPServerBase(MCPServerBase, ABC):
-    server: Any | None = None  # Using `Any` to avoid circular import issues
-    framework: Literal[AgentFramework.AGNO] = AgentFramework.AGNO
-
-    def _check_dependencies(self) -> None:
-        """Check if the required dependencies for the MCP server are available."""
-        self.libraries = "any-agent[mcp,agno]"
-        self.mcp_available = mcp_available
-        super()._check_dependencies()
+class AgnoMCPConnection(MCPConnection, ABC):
+    _server: AgnoMCPTools | None = PrivateAttr(default=None)
 
     @abstractmethod
-    async def _setup_tools(self) -> None:
-        """Set up the Agno MCP server with the provided configuration."""
-        if not self.server:
-            msg = "MCP server is not set up. Please call `setup` from a concrete class."
+    async def list_tools(self) -> list[Tool]:
+        """List tools from the MCP server."""
+        if self._server is None:
+            msg = "MCP server is not set up. Please call `list_tools` from a concrete class."
             raise ValueError(msg)
 
-        self.tools = [await self._exit_stack.enter_async_context(self.server)]
+        tools = await self._exit_stack.enter_async_context(self._server)
+        return tools if isinstance(tools, list) else [tools]  # type: ignore[return-value, list-item]
```

Comment:
Is there any instance where tools is a list? It's not clear to me why this logic is required

Reply:
I introduced this change to make one test pass which was using several patches, now it is no longer needed, I'm reverting it to a similar implementation as before